src/db.c

```diff
@@ -46,6 +48,80 @@ void updateLFU(robj *val) {
     val->lru = (LFUGetTimeInMinutes()<<8) | counter;
 }
 
+/* 
+ * Update histogram of keys-sizes
+ * 
+ * It is used to track the distribution of key sizes in the dataset. It is updated 
+ * every time key's length is modified. Available to user via INFO command. 
+ * 
+ * The histogram is a base-2 logarithmic histogram, with 64 bins. The i'th bin 
+ * represents the number of keys with a size in the range 2^i and 2^(i+1) 
+ * exclusive. oldLen/newLen must be smaller than 2^63. Each data type has its
+ * own histogram and it is per database (In addition, there is histogram per 
+ * slot for future use).
+ * 
+ * Examples to LEN values and corresponding bins in histogram: 
+ *               [1,2)->0 [2,4)->1 [4,8)->2 [8,16)->3
+ */
+void updateKeysizesHist(redisDb *db, int didx, uint32_t type, uint64_t oldLen, uint64_t newLen) {
+    uint64_t dummyHist[MAX_KEYSIZES_BINS];
+
+    if  (unlikely(type >= OBJ_TYPE_BASIC_MAX)) 
+        return;
+
+    kvstoreDictMetadata *metadata = kvstoreGetDictMetadata(db->keys, didx);
+
+    /* If following key deletion, If it is last one in slot's dict, then 
+     * slot's dict might get released as well. Verify if metadata is available. */
+    uint64_t *dictHist = (metadata != NULL) ? metadata->keysizes_hist[type] : dummyHist;
+
+    uint64_t *kvstoreHist = kvstoreGetMetadata(db->keys)->keysizes_hist[type];
+
+    if (oldLen != 0) {
+        int oLdBin = log2ceil(oldLen);
```

Comment:
```suggestion
        int oldBin = log2ceil(oldLen);
```
may you want to use oldBin?
maybe we can use underline naming method for variables more uniformly, just like most variables in redis.